src/libstore/log-store.hh

```diff
@@ -11,7 +11,14 @@ struct LogStore : public virtual Store
 
     /* Return the build log of the specified store path, if available,
        or null otherwise. */
-    virtual std::optional<std::string> getBuildLog(const StorePath & path) = 0;
+    std::optional<std::string> getBuildLog(const StorePath & path) {
+      auto maybePath = getBuildDerivationPath(path);
+      if (!maybePath)
+          return std::nullopt;
+      return getBuildLogExact(maybePath.value());
+    }
```

Comment:
Can you move the implementation to the `.cc` file? That makes the interface file easier to read and improves the build speed a tiny bit.

Reply:
Opened #7597 for that